### ISSUE TITLE
Inconsistent handling of singular non-message fields

### ISSUE THREAD

**Issue by mlubin** [JuliaIO/ProtoBuf.jl#130]
ProtoBuf.jl does not handle singular non-message fields consistently with the Python API.

From https://developers.google.com/protocol-buffers/docs/reference/python-generated#singular-fields-proto2:
> If foo is read when it is not set, its value is the default value for that field.

From https://developers.google.com/protocol-buffers/docs/reference/python-generated#singular-fields-proto3:
> If foo is read when it is not set, its value is the default value for that field.

> Unlike in proto2, you cannot call HasField() for a singular non-message field in proto3, and the library will throw an exception if you try to do this.

```proto
syntax = "proto2";

package proto2;

message Foo {
  optional double bar = 1;
}
```

```proto
syntax = "proto3";

package proto3;

message Foo {
  double bar = 1;
}
```

```julia
julia> get_field(proto2.Foo(), :bar)
ERROR: uninitialized field bar # The python API would return 0.0 here.

julia> get_field(proto3.Foo(), :bar)
ERROR: uninitialized field bar # This doesn't make sense because singular non-message fields have no "set" or "unset" status.

julia> isfilled(proto3.Foo(), :bar)
false # Correct

julia> isfilled(proto2.Foo(), :bar)
false # Should be an error or true.
```

**Comment by tanmaykm** [comment]
So, if I read it right, both proto2 and proto3 specs specify the same thing about reading values that were not set in the message.

I think the current behavior does confirm to that. The struct is initialized, and `isfilled` property gets set, only after the struct is read from a message buffer. The code above seems like doing that against a freshly constructed struct.

**Comment by tanmaykm** [comment]
Yes, I see that now.

**Comment by tanmaykm** [comment]
This seems to have been fixed since v0.9 after the code generator rework. Generating and loading the proto samples mentioned in this issue with the latest ProtoBuf.jl version:

```julia
julia> include("foo1_pb.jl")

julia> foo = readproto(IOBuffer(), Foo());

julia> foo.bar
0.0
```

```julia
julia> include("foo1_pb.jl")

julia> foo = readproto(IOBuffer(), Foo());

julia> foo.bar
0.0
```

Closing.
